consensus/voteset/voteset.go

```diff
@@ -4,174 +4,69 @@ import (
 	"fmt"
 
 	"github.com/pactus-project/pactus/crypto"
-	"github.com/pactus-project/pactus/crypto/bls"
-	"github.com/pactus-project/pactus/crypto/hash"
-	"github.com/pactus-project/pactus/types/block"
 	"github.com/pactus-project/pactus/types/validator"
 	"github.com/pactus-project/pactus/types/vote"
 	"github.com/pactus-project/pactus/util/errors"
 )
 
-type VoteSet struct {
-	round      int16
+type voteSet struct {
 	voteType   vote.Type
-	validators []*validator.Validator
-	blockVotes map[hash.Hash]*blockVotes
-	allVotes   map[hash.Hash]*vote.Vote
+	round      int16
+	validators map[crypto.Address]*validator.Validator
 	totalPower int64
-	quorumHash *hash.Hash
 }
 
-func NewVoteSet(round int16, voteType vote.Type, validators []*validator.Validator) *VoteSet {
-	totalPower := int64(0)
-	for _, val := range validators {
-		totalPower += val.Power()
-	}
-
-	return &VoteSet{
+func newVoteSet(voteType vote.Type, round int16, totalPower int64,
+	validators map[crypto.Address]*validator.Validator,
+) *voteSet {
+	return &voteSet{
 		round:      round,
 		voteType:   voteType,
 		validators: validators,
 		totalPower: totalPower,
-		blockVotes: make(map[hash.Hash]*blockVotes),
-		allVotes:   make(map[hash.Hash]*vote.Vote),
 	}
 }
 
-func (vs *VoteSet) Type() vote.Type { return vs.voteType }
-func (vs *VoteSet) Round() int16    { return vs.round }
-
-func (vs *VoteSet) Len() int {
-	return len(vs.allVotes)
+func (vs *voteSet) Type() vote.Type {
+	return vs.voteType
 }
 
-func (vs *VoteSet) AllVotes() []*vote.Vote {
-	votes := make([]*vote.Vote, 0)
-	for _, v := range vs.allVotes {
-		votes = append(votes, v)
-	}
-	return votes
+// Round returns the round number for the VoteSet.
+func (vs *voteSet) Round() int16 {
+	return vs.round
 }
 
-func (vs *VoteSet) getValidatorByAddress(addr crypto.Address) *validator.Validator {
-	for _, val := range vs.validators {
-		if val.Address().EqualsTo(addr) {
-			return val
-		}
-	}
-	return nil
-}
-
-func (vs *VoteSet) mustGetBlockVotes(blockhash hash.Hash) *blockVotes {
-	bv, exists := vs.blockVotes[blockhash]
-	if !exists {
-		bv = newBlockVotes()
-		vs.blockVotes[blockhash] = bv
-	}
-	return bv
-}
-
-func (vs *VoteSet) AddVote(v *vote.Vote) error {
-	if (v.Round() != vs.Round()) ||
-		(v.Type() != vs.Type()) {
-		return errors.Errorf(errors.ErrInvalidVote, "expected %d/%s, but got %d/%s",
-			vs.Round(), vs.Type(),
-			v.Round(), v.Type())
+// verifyVote checks if the given vote is valid.
+// It returns the voting power of if valid, or an error if not.
+func (vs *voteSet) verifyVote(v *vote.Vote) (int64, error) {
+	if v.Round() != vs.Round() ||
+		v.Type() != vs.Type() {
+		panic("remove me later")
```

Comment:
we should not delete `panic` from code?

Reply:
Good find. I had plan to remove this part. It missed.

Reply:
good, please remove and resolve conversation